### PR TITLE
Add JavaScript Node.js Console Lexer

### PR DESCRIPTION
**Pull request by cltrudeau**
I needed something like pycon for Node in a project I was working on, once written it was easy enough to put in here.

I've tested under py36 - py39. I don't have py35 or pypy installed, hopefully everything will work there.

Drop me a note if you need something formatted differently / more tests / whatever.

Thanks for the great tool!

### CODE REVIEW

File: pygments/lexers/javascript.py [hunk 1]
```diff
@@ -1534,3 +1536,63 @@ class JuttleLexer(RegexLexer):
...
+        undefined
+        > a
+        3
+        > let b = '4'
+        undefined
+        > b
+        '4'
+        > b == a
+        false
+    """
```
Comment (Anteru) [inline pygments/lexers/javascript.py:1557]: Please add a `.. versionadded: 2.10` line at the end.

File: pygments/lexers/javascript.py [hunk 2]
```diff
@@ -1534,3 +1536,63 @@ class JuttleLexer(RegexLexer):
...
+        for match in line_re.finditer(text):
+            line = match.group()
+            if line.startswith('> '):
+                insertions.append((len(curcode), 
+                    [(0, Generic.Prompt, line[:2])]))
+
+                curcode += line[2:]
+            elif line.startswith('...'):
+                # node does a nested ... thing depending on depth
+                code = line.lstrip('.')
```
Comment (Anteru) [inline pygments/lexers/javascript.py:1578]: If you know it's `...`, wouldn't `line [4:]` be faster here, or is it possible that the prefix part is longer?

**Review by Anteru — CHANGES_REQUESTED** [review]

File: pygments/lexers/javascript.py (hunk 2 above)
Comment (Mesteery) [reply pygments/lexers/javascript.py:1578]: > or is it possible that the prefix part is longer?

Yes
image

**Comment by Mesteery** [comment]
I find 'JavaScript Node.js console session' too ambiguous.
Wouldn't talking about 'REPL' and removing 'JavaScript' be better?
Also :
```
        aliases = ['nodejs', 'nodecon'] <- nodejs alias is false, nodecon is too ambiguous
        mimetypes = ['application/javascript', 'application/x-javascript', ...] <- these mime types are false
```

**Comment by Anteru** [comment]
> I find 'JavaScript Node.js console session' too ambiguous.

Can you please elaborate as to how this is ambiguous? What would you suggest? `nodejsrepl`?

**Comment by Mesteery** [comment]
> Can you please elaborate as to how this is ambiguous? What would you suggest? nodejsrepl?

Yes, something like nodejsrepl.

It's not really ambiguous, but it would be better to have a reference to 'REPL', without 'JavaScript'.

For example: 'Node.js REPL session'.

**Comment by cltrudeau** [comment]
> I find 'JavaScript Node.js console session' too ambiguous.
> Wouldn't talking about 'REPL' and removing 'JavaScript' be better?
> Also :
> 
> ```
>         aliases = ['nodejs', 'nodecon'] <- nodejs alias is false, nodecon is too ambiguous
>         mimetypes = ['application/javascript', 'application/x-javascript', ...] <- these mime types are false
> ```

Alias has been fixed, mimetypes have been reduced to text/x-nodejsrepl.

I tried renaming the Lexer to NodeREPLLexer for consistency with this suggestion, but it breaks the tests cases -- the have a special condition in them for things named Console that seems to be needed here. I didn't dig any further, just put it back to NodeConsoleLexer.

**Review by Anteru — APPROVED** [review]

**Comment by Anteru** [comment]
Merged, thanks for your contribution!

**Pull request merged**